src/__experimental__/components/select/select.component.js

```diff
@@ -70,17 +88,85 @@ class Select extends React.Component {
 
   handleClick = () => this.openList()
 
-  handleChange = (newValue) => {
-    let updatedValue = newValue;
+  /**
+   * Is this component currently controlled (by the `value` prop)?
+   */
+  isComponentControlled() {
+    return (this.props.value !== undefined && this.props.value !== null);
+  }
```

Comment:
do we need to assume that this component could change from uncontrolled to controlled?
couldn't it be just `isComponentControlled = this.props.value !== undefined && this.props.value !== null;`?

Reply:
You're correct. In fact @edleeks87 created FE-2262 to cover this for Carbon in general. I've now modified `isComponentControlled` in 855908d :thumbsup: